### PR TITLE
RMWC 6

### PR DESCRIPTION
**Pull request by tstirrat**
Wow, material-components-web and SASS modules - what a nightmare. 🙀

Was required to change to the dart sass compiler (slower) because `@material/*` sass files need it. But now we have `@use` support.

Regression in select field tab indexes, so I've kept `@rmwc/select` at 5.7.2. Raised in RMWC: https://github.com/jamesmfriedman/rmwc/issues/595

There are some act/async warnings with TabScroller/TabIndicator animations.. I'm not sure on the right move here, perhaps a fix in the underlying components.. but for now. I added 9000 awaits into the tests... sorry.. This was the only thing that worked from many many attempts.

Ive run the tests a bunch of times to make sure they're not flaky.

### CODE REVIEW

File: src/components/App/AppBar.test.tsx [hunk 1]
```diff
@@ -34,17 +34,36 @@ it('selects the matching nav-tab', () => {
...
+  await wait();
+  await wait();
+  await wait();
+  await wait();
+  await wait();
+  await wait();
+  await wait();
+  await wait();
+  await wait();
+  await wait(); // dont ask. Due to TabScroller doing stuff on requestAnimationFrame
```
Comment (tstirrat) [inline src/components/App/AppBar.test.tsx:65]: unfortunately, other things like `await wait(() => {}, {timeout: 10000})` did not work.

File: src/components/common/_material-overrides-inputs.scss [hunk 2]
```diff
@@ -14,49 +14,87 @@
...
+  $input-padding-top: $input-default-padding,
+  $input-padding-bottom: $input-default-padding,
+  $label-offset: $input-default-padding,
+  $outlined-idle-border: var(--fire-color-input-borders),
+  $outlined-hover-border: var(--fire-color-input-borders),
+  $outlined-input-padding-top: $input-default-padding,
+  $outlined-input-padding-bottom: $input-default-padding,
+);
+
+// TODO: uncomment for 6.0.9, when fixed
```
[the remark this answers is not in the document]
Comment (tstirrat) [reply src/components/common/_material-overrides-inputs.scss:33]: Done

File: src/components/common/utils.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * UUID using web crypto.
+ *
+ * @link https://gist.github.com/jed/982883#gistcomment-45104
+ */
+export function uuid() {
```
[the remark this answers is not in the document]
Comment (yuchenshi) [reply src/components/common/utils.ts:22]: I'm unresolving this -- I don't actually care about it that much, but unfortunately it's against Google policy to use WTFPL code. So please make sure this is addressed before you merge so we don't get yelled at.

[the remark this answers is not in the document]
Comment (tstirrat) [reply src/components/common/utils.ts:22]: Fixed in #158

**Pull request merged**